src/liboslexec/llvm_util.cpp

```diff
@@ -666,13 +648,19 @@ LLVM_Util::detect_cpu_features(TargetISA requestedISA, bool no_fma)
         return false;  // Could not figure it out
     }
 
-    bool target_requested = (requestedISA != TargetISA::UNKNOWN);
+    // Try to match features to the combination of the requested ISA and
+    // what the host CPU is able to support. Note that we intentionally
+    // disable AVX512 ISA support on AVX512 hardware unless LLVM >= 8.0,
+    // to avoid some performance issues that we can't fix without LLVM 8+
+    // features.
```

Comment:
Suggest not limiting here, but in backend when choosing a targetISA

Reply:
Aha, gotcha. I'm going to keep it extra simple, and not try to check or enforce this, since our current guidance is to not target 256 or 512 bit ops when doing single point shading, regardless. We'll measure separately for batch shading when we merge it, and it may be that at such time the guidance we give is that people should always use LLVM >= 8.0 if they want to employ the batch shading.